nano/core_test/election_scheduler.cpp

```diff
@@ -0,0 +1,126 @@
+	auto send1 = builder.make_block ()
+				 .account (nano::dev_genesis_key.pub)
+				 .previous (nano::genesis_hash)
+				 .representative (nano::dev_genesis_key.pub)
+				 .balance (nano::genesis_amount - nano::Gxrb_ratio)
+				 .link (nano::dev_genesis_key.pub)
+				 .sign (nano::dev_genesis_key.prv, nano::dev_genesis_key.pub)
+				 .work (*system.work.generate (nano::genesis_hash))
+				 .build_shared ();
+	system.nodes[0]->ledger.process (system.nodes[0]->store.tx_begin_write (), *send1);
+	system.nodes[0]->scheduler.activate (nano::dev_genesis_key.pub, system.nodes[0]->store.tx_begin_read ());
+	ASSERT_TIMELY (1s, system.nodes[0]->active.election (send1->qualified_root ()));
+}
+
+TEST (election_scheduler, activate_one_flush)
+{
+	nano::system system{ 1 };
+	nano::state_block_builder builder;
+	auto send1 = builder.make_block ()
+				 .account (nano::dev_genesis_key.pub)
+				 .previous (nano::genesis_hash)
+				 .representative (nano::dev_genesis_key.pub)
+				 .balance (nano::genesis_amount - nano::Gxrb_ratio)
+				 .link (nano::dev_genesis_key.pub)
+				 .sign (nano::dev_genesis_key.prv, nano::dev_genesis_key.pub)
+				 .work (*system.work.generate (nano::genesis_hash))
+				 .build_shared ();
+	system.nodes[0]->ledger.process (system.nodes[0]->store.tx_begin_write (), *send1);
+	system.nodes[0]->scheduler.activate (nano::dev_genesis_key.pub, system.nodes[0]->store.tx_begin_read ());
+	system.nodes[0]->scheduler.flush ();
+	ASSERT_NE (nullptr, system.nodes[0]->active.election (send1->qualified_root ()));
+}
+
+TEST (election_scheduler, no_vacancy)
+{
+	nano::system system;
+	nano::node_config config{ nano::get_available_port (), system.logging };
+	config.active_elections_size = 1;
+	auto & node = *system.add_node (config);
+	;
+	nano::state_block_builder builder;
+	nano::keypair key;
+
+	// Activating accounts depends on confirmed dependencies. First, prepare 2 accounts
+	auto send = builder.make_block ()
+				.account (nano::dev_genesis_key.pub)
+				.previous (nano::genesis_hash)
+				.representative (nano::dev_genesis_key.pub)
+				.link (key.pub)
+				.balance (nano::genesis_amount - nano::Gxrb_ratio)
+				.sign (nano::dev_genesis_key.prv, nano::dev_genesis_key.pub)
+				.work (*system.work.generate (nano::genesis_hash))
+				.build_shared ();
+	auto receive = builder.make_block ()
+				   .account (key.pub)
+				   .previous (0)
+				   .representative (key.pub)
+				   .link (send->hash ())
+				   .balance (nano::Gxrb_ratio)
+				   .sign (key.prv, key.pub)
+				   .work (*system.work.generate (key.pub))
+				   .build_shared ();
+	ASSERT_EQ (nano::process_result::progress, node.process (*send).code);
+	ASSERT_EQ (nano::process_result::progress, node.process (*receive).code);
+	node.block_confirm (send);
+	auto election1 = node.active.election (send->qualified_root ());
+	election1->force_confirm ();
+	node.block_confirm (receive);
+	auto election2 = node.active.election (receive->qualified_root ());
+	election2->force_confirm ();
```

Comment:
Probably can be replaces with `nano::blocks_confirm (node, { send, receive }, true);`